src/gc.c

```diff
@@ -108,7 +108,7 @@ static void run_finalizer(jl_ptls_t ptls, jl_value_t *o, jl_value_t *ff)
     JL_TRY {
         size_t last_age = jl_get_ptls_states()->world_age;
         jl_get_ptls_states()->world_age = jl_world_counter;
-        jl_apply(args, 2);
+        jl_apply_with_saved_exception_state(args, 2, 0);
         jl_get_ptls_states()->world_age = last_age;
     }
     JL_CATCH {
```

Comment:
We should either remove this or keep it working. (Printing a warning if an unexpected error happens in a finalizer is probably still somewhat useful.)